### PR TITLE
Adding encrypted PVC expansion test automation when PVC is 100% full

### PR DESCRIPTION
**Pull request by paraggit**
Test Steps:
            1. Create a encrypted storageclass.
            2. Create a encrypted PVC of size 4G using defined vault configuration.
            3. create a pod and mount the PVC inside the pod.
            4. Fill FS 100% using FIO workload.
            5. Expand PVC to size 10G
            6. continue IO workload and verify the IOs are succeeding
            7. Verify utilization alert will stop firing after volume expansion.

### CODE REVIEW

File: tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        kv_version,
+        kms_provider,
+        pv_encryption_kms_setup_factory,
+        use_vault_namespace,
+        project_factory,
+        storageclass_factory,
+        create_pvcs_and_pods,
+    ):
+        """
+        Setup csi-kms-connection-details configmap
```
Comment (ebenahar) [inline tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py:85]: this fixture does more than that

File: tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+                jobs=8,
+                size=f"{self.pvc_size}G",
+                io_direction="write",
+                fio_filename=f"{pod_obj.name}_f1",
+            )
+        log.info("Started IO on all to utilise 100% of PVCs")
+        # Wait for IO to finish
+        log.info("Wait for IO to finish on pods")
+        for pod_obj in self.pods:
+            # for pod_obj in self.pod_objs:
```
Comment (ebenahar) [inline tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py:149]: should be deleted I guess

File: tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+            try:
+                pod_obj.get_fio_results()
+            except CommandFailed as cfe:
+                if "No space left on device" not in str(cfe):
+                    raise
```
Comment (ebenahar) [inline tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py:154]: isn't it expected that the device will run out of space if we fill it to 100% capacity? If so, why is the exception being raised and not caught?

File: tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+                    alert
+                    for alert in alerts
+                    if alert.get("labels", {}).get("persistentvolumeclaim")
+                    == pvc_obj.name
+                ]
+                # At least 2 alerts should be present
+                if len(alerts_pvc) < 2:
+                    break
+
+                # Verify 'PersistentVolumeUsageNearFull' alert is firing
```
Comment (shyRozen) [inline tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py:180]: Is there a reason to check both PersistentVolumeUsageNearFull (75%) and PersistentVolumeUsageCritical (85%). Just trying to understand why use both as we try to exhaust the capacity, maybe checking 85% alert is enough?

File: tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+            5. Expand PVC to size 10G
+            6. continue IO workload and verify the IOs are succeeding
+            7. Verify utilization alert will stop firing after volume expansion.
+
+        """
+        pvc_size_expanded = 10
+
+        # Run IO to utilise 100% of volume
+        log.info("Run IO on all to utilise 100% of PVCs")
+        for pod_obj in self.pods:
```
Comment (shyRozen) [inline tests/functional/pv/pv_encryption/test_encrypted_pvc_expansion_when_full.py:135]: If you'd like you can make them run parallel and save some time. Here is an example [https://github.com/paraggit/ocs-ci/blob/377ff1d814ba702eb1d870570f1ada12198cf654/tests/lvmo/test_lvm_multi_clone.py#L136-L162](https://github.com/paraggit/ocs-ci/blob/377ff1d814ba702eb1d870570f1ada12198cf654/tests/lvmo/test_lvm_multi_clone.py#L136-L162)

**Pull request closed without merging**